src/App.tsx

```diff
@@ -1,26 +1,170 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import {
+  USER_ID,
+  getTodos,
+  createTodo,
+  deleteTodo,
+  updateToDo,
+} from './api/todos'; // Ensure createTodo is imported
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { Error } from './components/Error/Error';
 
-const USER_ID = 0;
+function getVisibleToDos(newTodos: Todo[], newStatus: Status) {
+  switch (newStatus) {
+    case Status.Active:
+      return newTodos.filter(todo => !todo.completed);
+
+    case Status.Completed:
+      return newTodos.filter(todo => todo.completed);
+
+    default:
+      return newTodos;
+  }
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState('');
+  const [status, setStatus] = useState(Status.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [title, setTitle] = useState('');
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setError('Unable to load todos');
+      });
+  }, []);
+
+  useEffect(() => {
+    let timeoutId: NodeJS.Timeout;
+
+    if (error) {
+      timeoutId = setTimeout(() => setError(''), 3000);
+    }
+
+    return () => clearTimeout(timeoutId);
+  }, [error]);
+
+  const visibleToDos = useMemo(
+    () => getVisibleToDos(todos, status),
+    [todos, status],
+  );
+
+  const addTodo = (toDoTitle: string) => {
+    const newTitle = toDoTitle.trim();
+
+    if (!newTitle) {
+      setError('Title should not be empty');
+
+      return;
+    }
+
+    setIsLoading(true);
+
+    const newTodo = { id: 0, title: newTitle, completed: false };
+
+    setTempTodo(newTodo);
+
+    return createTodo(newTodo)
+      .then(resultTodo => {
+        setTodos([...todos, resultTodo]);
```

Comment:
It's better to access the state inside the function to prevent call overlapping:
```
setTodos((tds) => [...tds, resultTodo])
```